src/Rules/Classes/ClassExtendsInternalClassRule.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+
+namespace mglaman\PHPStanDrupal\Rules\Classes;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+
+class ClassExtendsInternalClassRule implements Rule
+{
+    /**
+     * @var ReflectionProvider
+     */
+    private $reflectionProvider;
+
+    public function __construct(ReflectionProvider $reflectionProvider)
+    {
+        $this->reflectionProvider = $reflectionProvider;
+    }
+
+    public function getNodeType(): string
+    {
+        return Class_::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        /** @var Class_ $node */
+        if (!isset($node->extends)) {
+            return [];
+        }
+
+        $extendedClassName = $node->extends->toString();
+        if (!$this->reflectionProvider->hasClass($extendedClassName)) {
+            return [];
+        }
+
+        $extendedClassReflection = $this->reflectionProvider->getClass($extendedClassName);
+        if (!$extendedClassReflection->isInternal()) {
+            return [];
+        }
+
+        if (!isset($node->namespacedName)) {
+            return $this->buildError(null, $extendedClassName);
+        }
```

Comment:
Should we error here? Some code in PHPStan shows it throws a ShouldNotHappenException 

https://github.com/phpstan/phpstan-src/search?q=namespacedName

Reply:
We should not error here because it handles anonymous classes.

PHPStan throws for interface, trait, function etc.. or [when an anonymous class has a namespaced name](https://github.com/phpstan/phpstan-src/blob/629ccf6b02146e1cbf0499982358c595e3d6edd5/src/Reflection/BetterReflection/BetterReflectionProvider.php#L205)